### PR TITLE
Add --implicit option to :open

### PR DESCRIPTION
**Pull request by mlochbaum**
Some keybindings with `:open -t` or `:open -b` open a tab which is related to the current one. In particular, `hint links fill :open -t {hint-url}` (default `;O`) essentially opens a link with the ability to inspect it first. Adding an `--implicit` flag to `:open`, which treats any new tab opened by the command as implicitly opened, lets these commands work like opening a link does.

### CODE REVIEW

File: tests/end2end/features/open.feature [hunk 1]
```diff
@@ -108,3 +108,37 @@ Feature: Opening pages
...
+        And I wait until data/numbers/9.txt is loaded
+        Then the session should look like:
+            windows:
+            - tabs:
+              - history:
+                - url: about:blank
+              - active: true
+                history:
+                - active: true
+                  url: http://localhost:*/data/numbers/9.txt
```
Comment (The-Compiler) [inline tests/end2end/features/open.feature]: You should be able to use "Then the following tabs should be open:" (see `tabs.feature`) to simplify this a bit (I probably should do the same for other tests in here...)

File: qutebrowser/config/configdata.py [hunk 2]
```diff
@@ -1656,4 +1656,9 @@ def data(readonly=False):
     (re.compile(r'^download-remove --all$'), r'download-clear'),
 
     (re.compile(r'^hint links fill "([^"]*)"$'), r'hint links fill \1'),
+
+    (re.compile(r'^set-cmd-text :open -([tb]) {url:pretty}$'),
+                 r'set-cmd-text :open -\1 -i {url:pretty}'),
+    (re.compile(r'^hint links fill :open -t {hint-url}$'),
+                 r'hint links fill :open -t -i {hint-url}'),
```
Comment (The-Compiler) [inline qutebrowser/config/configdata.py]: Hmm, I'm not really comfortable about this. Those fixes are mainly for things which are clearly not valid anymore, to automatically fix them in user's configs. Having this here would mean an user can never remove `-i` from those keybindings.... So I'd rather not have this rebinding done automatically.

(I know this sucks... I want to rip out the current config system as soon as the QtWebEngine-stuff is over)

File: tests/end2end/features/open.feature (hunk 1 above)
Comment (mlochbaum) [inline tests/end2end/features/open.feature]: I tried that before and it didn't work, but I see now that all of my failures were due to placing the tests after the one that opened a new window, and the final success was just from running with `--lf`. I'm rewriting the other tests now. Is there a way to test multiple windows with this syntax?

File: qutebrowser/config/configdata.py (hunk 2 above)
Comment (mlochbaum) [inline qutebrowser/config/configdata.py]: That all makes sense. I've been gearing up to join the config revolution for a while.

**Comment by The-Compiler** [comment]
Regarding "Then the following tabs should be open:" and multiple windows: It deliberately checks that there's only one window open.

Ideally there would be a `:window-only` command which closes all windows except the current one (like `:tab-only`). Then `quteprocess.py` would take care of running that after every test. That'd also mean we could remove `Given I have a fresh instance` for various tests, which would make them a bit faster. However, I didn't get to doing that yet... Maybe something for you? :wink:

**Comment by The-Compiler** [comment]
I just noticed this in the middle of merging, but what do you think about adding an error message if `--implicit` is given without `--tab` or `--bg`?

**Comment by mlochbaum** [comment]
Maybe open it and give a warning? In the future we might care about whether a tab in a new window was opened explicitly or implicitly, but that wouldn't be handled correctly with the current code anyway.

I'll look at writing a `:window-only`.

**Comment by The-Compiler** [comment]
I'd say let's rather silently ignore it for now then :wink: Thanks!

**Pull request merged**